### ISSUE TITLE
InternalOAuthError: Failed to fetch user profile

### ISSUE THREAD

**Issue by nicutimofte** [fh1ch/passport-gitlab2#9]
After I authorize the project to have Gitlab access I got this error:

![image](https://user-images.githubusercontent.com/18091830/41199810-4ff9fa64-6ca1-11e8-97e2-62a889f7f7a0.png)

Implementation:
middleware
```
passport.use(new gitlab.Strategy({
                clientID: config.GITLAB.APPLICATION_ID,
                clientSecret: config.GITLAB.SECRET,
                callbackURL: "http://localhost:3000/auth/gitlab/callback"
            }, (accessToken, refreshToken, profile, cb) => {
                console.log(accesstoken)
                return cb(null, profile)
            }
        ))
```
auth route
```
 passport.authenticate('gitlab', {
        state: userId,
        scope: ['api']
    })
```

callback
```

router.get('/auth/gitlab/callback',
    passport.authenticate('gitlab', {
        failureRedirect: '/login'
    }), gitlabCallback )
```
I have to mention that authentication was working until a couple of weeks ago, when I worked last time on the project and now this error occurred. Moreover, the code and the necessary tokens were not changed since implementation. It might be a change in Gitlab's api or am I am missing something?

**Comment by fh1ch** [comment]
@nicutimofte thanks for the bug report. What version of GitLab are you using?

Could you please update to the newly released `passport-gitlab2@4.0.0` and verify it again? There was an update to utilize the GitLab APIv4.

**Comment by nicutimofte** [comment]
I was sure I checked the last available version and it was `3.0.0`. But it seems I was wrong. 
It worked after I updated to `4.0.0`. Thank you, @fh1ch!

**Comment by fh1ch** [comment]
Well, `4.0.0` was just released yesterday 😄 Glad that it solves your problem!

Cheers
